scripts/solve_network.py

```diff
@@ -77,6 +77,22 @@ def bau_mincapacities_rule(model, carrier):
         ext_gens_i = n.generators.index[n.generators.carrier.isin(conv_techs) & n.generators.p_nom_extendable]
         n.model.safe_peakdemand = pypsa.opt.Constraint(expr=sum(n.model.generator_p_nom[gen] for gen in ext_gens_i) >= peakdemand - exist_conv_caps)
 
+def add_country_carrier_generation_constraints(n, opts=None):
+    agg_p_nom_minmax = pd.read_csv("data/agg_p_nom_minmax.csv", index_col=list(range(2)))
```

Comment:
Two remarks:
* This will fail if `data/agg_p_nom_minmax.csv` does not exist. There should be either
(a) a check that the file exists, and if not ensure that some warning is given and a default behaviour is implemented (e.g. limits are 0 and full potential of a country?) or
(b) you should raise a descriptive error message if the file could not be loaded.
* It might also make sense to specify the path to `agg_p_nom_minmax.csv` in the `config.yaml` to store multiple different limits in `./data`.

```python
agg_p_nom_limits = snakemake.config['electricity']['agg_p_nom_limits']
if os.path.isfile(agg_p_nom_limits):
  agg_p_nom_minmax = pd.read_csv(agg_p_nom_limits, index_col=list(range(2)))
else:
  raise FileNotFoundError("Need to specify the path to a .csv file containing aggregate capacity limits per country in config['electricity']['agg_p_nom_limit'].")
```

Reply:
If you follow this path, can you also amend the `config.yaml`?

```yaml
electricity:
  agg_p_nom_limits: data/agg_p_nom_minmax.csv
```

https://github.com/PyPSA/pypsa-eur/blob/master/config.yaml#L27-L37

Reply:
I included your changes, thanks!

Reply:
You're quick. I just thought this would be a better error message:
```
FileNotFoundError("{} not found. \
    Need to specify the path to an existing .csv file \
    containing aggregate capacity limits per country \
    in config['electricity']['agg_p_nom_limit'].".format(agg_p_nom_limits))
```

Reply:
> You're quick. I just thought this would be a better error message:
> 
> ```python
> FileNotFoundError("{} not found. \
>     Need to specify the path to an existing .csv file \
>     containing aggregate capacity limits per country \
>     in config['electricity']['agg_p_nom_limit'].".format(agg_p_nom_limits))
> ```

In don't see any difference between this error message and the previous one you wrote.